### PR TITLE
Vijayasree / convert layout/footer folder and rest of the files in layout to TS

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

-    convert footer folder to typescript

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/custom/_eu-redirect-modal.tsx [hunk 1]
```diff
@@ -8,6 +7,17 @@ import device from 'themes/device'
...
 
+type EURedirectProps = {
+    aria_label?: string
+    closeModal?: () => void
+    is_open: boolean
+    ref?: Ref<HTMLAnchorElement>
+    rel?: string
+    target?: string
+    to?: string
+    toggle: () => void
```
Comment (michio-rework) [inline src/components/custom/_eu-redirect-modal.tsx:18]:
```suggestion
    toggle: React.MouseEventHandler<HTMLDivElement> | React.MouseEventHandler<HTMLImageElement>
```
what do you thing about this approach? 
or 
```suggestion
    toggle: React.MouseEventHandler<HTMLElement>
```
or
```suggestion
    toggle: (event:  React.MouseEvent<HTMLElement>) => void 
```

File: src/components/layout/footer/social-wrapper.tsx [hunk 2]
```diff
@@ -18,7 +17,27 @@ import Instagram from 'images/svg/layout/footer-instagram.svg'
...
 
-const SocialWrapperComponent = ({ is_career_page, fb_url, instagram_url, twitter_url }) => {
+type SocialWrapperComponentProps = {
+    fb_url?: string
+    instagram_url?: string
+    is_career_page?: boolean
+    twitter_url?: string
+}
+
+type SocialMediaComponentProps = {
```
Comment (michio-rework) [inline src/components/layout/footer/social-wrapper.tsx:27]:
```suggestion
   type TSocialAccount = {
        link?: string
        image?: string
        image_alt?: string
    }
    type SocialMediaComponentProps = {
            social_accounts: TSocialAccount[]
    }
```

File: src/components/layout/location-context.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React, { ReactNode, Dispatch, SetStateAction } from 'react'
+import type { ModalPayloadType } from './layout'
+
+type LocationProviderProps = {
+    children: ReactNode
+    has_mounted?: boolean
+    setModalPayload?: Dispatch<SetStateAction<ModalPayloadType>>
+    show_cookie_banner?: boolean
+    toggleModal?: () => void
```
Comment (michio-rework) [inline src/components/layout/location-context.tsx:9]:
```suggestion
    toggleModal?: React.MouseEventHandler<HTMLElement>
```
or something like this, I mean we should provide some kinda type for the event, we may need it in other components. :pray:

**Review by michio-rework — CHANGES_REQUESTED** [review]

File: src/components/custom/_eu-redirect-modal.tsx (hunk 1 above)
Comment (NikitK-deriv) [reply src/components/custom/_eu-redirect-modal.tsx:18]: Cool idea @michio-rework 👍 
Let my try to implement it

Comment (NikitK-deriv) [reply src/components/custom/_eu-redirect-modal.tsx]: Please check again 🙏

**Review by michio-rework — DISMISSED** [review]

**Review by mohammad-rework — APPROVED** [review]

**Review by mohammad-rework — APPROVED** [review]

**Pull request merged**